src/Processors/Formats/Impl/AvroRowInputFormat.cpp

```diff
@@ -153,6 +157,39 @@ static void insertNumber(IColumn & column, WhichDataType type, T value)
     }
 }
 
+template <typename DecimalType>
+static AvroDeserializer::DeserializeFn createDecimalDeserializeFn(const avro::NodePtr & root_node, const DataTypePtr & target_type)
+{
+    auto logical_type = root_node->logicalType();
+    const auto & decimal_type = assert_cast<const DecimalType &>(*target_type);
+    if (decimal_type.getScale() != static_cast<UInt32>(logical_type.scale()) || decimal_type.getPrecision() != static_cast<UInt32>(logical_type.precision()))
+        throw Exception(
+            ErrorCodes::BAD_ARGUMENTS,
+            "Cannot insert Avro decimal with scale {} and precision {} to ClickHouse Decimal with scale {} and precision {}",
+            logical_type.scale(),
+            logical_type.precision(),
+            decimal_type.getScale(),
+            decimal_type.getPrecision());
+
+    return [tmp = std::string(), target_type](IColumn & column, avro::Decoder & decoder) mutable
```

Comment:
why do we do this with `tmp`?
Why not create a temp variable on each call?

Reply:
yes it makes sense, I thought I saw move from it somewhere but it seems I'm crazy